asynciterator.ts

```diff
@@ -1457,9 +1469,8 @@ export class UnionIterator<T> extends BufferedIterator<T> {
     @param {object} [options] Settings of the iterator
   */
   constructor(sources: AsyncIteratorOrArray<AsyncIterator<T>>,
-              options: BufferedIteratorOptions = {}) {
-    super(options);
-    const autoStart = options.autoStart !== false;
+              { autoStart = false, ...options }: BufferedIteratorOptions = {}) {
+    super({ autoStart, ...options });
```

Comment:
I wouldn't deconstruct here; we can just use `options.autoStart` below. The original was needed because we didn't want to repeat the `!== false` cast, which is no longer needed (as in 6f3af334c76e792e674ffce3dd74b211ebb79308).